statics/js/components/preference.js

```diff
@@ -0,0 +1,98 @@
+/* jshint multistr: true */
+
+var PreferenceComponent = {
+
+  name: 'Preference',
+
+  mixins: [notificationMixin],
+
+  template: '\
+    <form class="form-preference" @submit.prevent="save">\
+      <div class="form-group">\
+        <label for="collapse-index">Collapse Index</label>\
+        <a><i class="fa fa-question help-text" aria-hidden="true" title="initial collapse value"></i></a>\
+        <input id="collapse-index" type="number" class="form-control input-sm" v-model="collapseIndex" min="0" />\
+      </div>\
+      <div class="form-group">\
+        <label for="default-filter">Filters</label>\
+        <a><i class="fa fa-question help-text" aria-hidden="true" title="filter and highlight gremlin queries, displayed in the left panel"></i></a>\
+        <div v-for="filter in filters">\
+          <div class="form-group">\
+            <div class="input-group">\
+              <label class="input-group-addon">Name: </label>\
+              <input class="form-control" v-model="filter.name"/>\
+            </div>\
+            <div class="input-group filter-field">\
+              <label class="input-group-addon">Filter:  </label>\
+              <input class="form-control" v-model="filter.filter"/>\
+            </div>\
+          </div>\
+        </div>\
+        <button class="btn btn-primary btn-round-xs btn-xs" type="button" @click="addFilter">+</button>\
+      </div>\
+      <div class="form-group">\
+        <label for="bw-threshold">Bandwidth Threshold</label>\
+        <a><i class="fa fa-question help-text" aria-hidden="true" title="bandwidth threshold mode (relative/absolute)"></i></a>\
+        <input id="bw-threshold" type="text" class="form-control input-sm" v-model="bwThreshold"/>\
+      </div>\
+      <div class="button-holder row">\
+        <button class="btn btn-lg btn-primary" type="submit">Save</button>\
+        <button class="btn btn-lg btn-danger" type="button" @click="cancel"> Cancel</button>\
+      </div>\
+    </form>\
+  ',
+
+  data: function() {
+    var ci = 0;
+    var filters = [{name:"", filter:""}];
+    var bwThreshold = "";
+    if (typeof(Storage) !== "undefined") {
+      ci = Number(localStorage.getItem("collapseIndex"));
+      f = JSON.parse(localStorage.getItem("filters"));
+      if (f && f.length > 0) filters = f;
+      bwThreshold = localStorage.getItem("bandwidthThreshold");
+    }
+    return {
+      collapseIndex: ci,
+      filters: filters,
+      bwThreshold: bwThreshold,
+    };
+  },
+
+  methods: {
+
+    save: function() {
+      if (this.bwThreshold !== "" && this.bwThreshold !== "absolute" && this.bwThreshold !== "relative") {
```

Comment:
we need to handle all the BW configuration parameters

Reply:
done